### PR TITLE
White board dash

### PR DESCRIPTION
**Pull request by Manik-Coder**
Able to Add Course In the database and View Courses..Trying to create Assignment link inside the course so that Assignment can be uploaded in Each Course .

### CODE REVIEW

**Comment by ericgeil223** [comment]
I am working on adding assignments. I have to do something in this course otherwise I won't get a grade.

**Comment by Manik-Coder** [comment]
Oh sure !! Do what you need to do.thanks

File: src/components/Courses.js [hunk 1]
```diff
@@ -1,8 +1,83 @@
...
+     <br/> 
+    <h3>Add Courses:</h3>
+    <form>
+      <input type="text" placeholder="Title" name="title" onChange={newCourseChanged} /><br/>
+      <input type="text" placeholder="Course Code" name="course_code" onChange={newCourseChanged} /><br/>
+      <input type="text" placeholder="Session Code" name="session_code" onChange={newCourseChanged} /><br/>
+      <input type="text" placeholder="Semester Code" name="semester_code" defaultValue="0" onChange={newCourseChanged} /><br/><br/>
+      <button className="btn btn-secondary" onClick={async (e) => {
+        e.preventDefault();
+        console.log("User created with ID: " + await CourseService.createCourse(newCourse));
```
Comment (n-grounds) [inline src/components/Courses.js]: not that we want console.log statements sporadically logging information in the finished product, but having some for the time being is fine, though I suspect this should be "Course created with ID" and not "User created ..."

File: src/components/Courses.js [hunk 2]
```diff
@@ -1,8 +1,83 @@
...
+    }>Display All Courses</button></Col>
+    </Row>
+    <br/>
+    <Row>
+    {courses.map((course) => {
+      return <Col><Card style={{width:'30rem' }}>
+        {/* <Card.Img variant="top" src="https://via.placeholder.com/30x30" /> */}
+        <Card.Body>
+          <Card.Title>Course Name</Card.Title>
+          <Card.Text><CourseDetails key={course.id} course={course} />
```
Comment (n-grounds) [inline src/components/Courses.js]: what exactly does the `key={course.id}` portion of this do?  it doesn't appear that in the `CourseDetails` function below that the `key` is used, like the `course` property is

**Review by n-grounds — APPROVED** [review]
Approved but with some suggestion for improvement and a question about how/why something is the way it is (mostly just so I understand)

File: src/components/Courses.js (hunk 2 above)
Comment (Manik-Coder) [reply src/components/Courses.js]: To our services to work key={course.id} is not required just to make it work with future enhancement like  selecting the course details or selecting the exact assignments we may need it .

File: src/components/Courses.js (hunk 1 above)
Comment (Manik-Coder) [reply src/components/Courses.js]: That was a Typo as I used the same services as that of user .The console.log was used for demo that won't be on the production.

File: src/components/Courses.js [hunk 3]
```diff
@@ -1,10 +1,95 @@
...
+          homework: "test",
+          homeworkSolution: "Solved",
+          quiz: "pass",
+          });
+
+  return <Container><div><h1>Courses page</h1>
+    <Row><Col>
+      <button type="button" className="btn btn-primary" onClick={
+        async () => {
+          setCourses(await CourseService.getCourses());
```
Comment (logan-giese) [inline src/components/Courses.js:35]: You can actually do this in a React hook (specifically useEffect) to load it automatically when the page is loaded.

File: src/index.css [hunk 4]
```diff
@@ -1,5 +1,6 @@
 body {
   margin: 0;
+ /* background-color: #610a15;*/
```
Comment (logan-giese) [inline src/index.css:3]: Not sure why this was added if it's just commented out anyway :/

**Review by logan-giese — APPROVED** [review]
It's a solid start for the courses page.  There are some improvements that could be made, but it's a great foundation for future work.  I'd recommend looking into making it so that the modal popups don't make the background go black, though.

File: src/index.css (hunk 4 above)
Comment (Manik-Coder) [reply src/index.css:3]: Actually i was testing it to make it look in different color.

File: src/components/Courses.js (hunk 3 above)
Comment (Manik-Coder) [reply src/components/Courses.js:35]: Noted that .

**Pull request merged**